scope_registry.go

```diff
@@ -86,19 +86,27 @@ func (r *scopeRegistry) Subscope(parent *scope, prefix string, tags map[string]s
 		return NoopScope.(*scope)
 	}
 
-	key := scopeRegistryKey(prefix, parent.tags, tags)
-
+	preSanitizeKey := keyForPrefixedStringMapsAsKey(prefix, parent.tags, tags)
```

Comment:
If you were to inline the `keyForPrefixedStringMapsAsKey` you would not need the object pool for the `bytes.Buffer`, but it does make this function longer. I'll leave it up to you to try it out (and decide what to do).

Reply:
yes, this was the reason I kept it as object pool instead of trying your suggestion. If I have to make byte slice on the stack, the method has to be inlined for it to work. However the same method is used from other places as well and it would need duplication of logic.